core/types/access_list_tx.go

```diff
@@ -270,11 +250,11 @@ func (tx *AccessListTx) encodePayload(w io.Writer, b []byte, payloadSize, nonceL
 		return err
 	}
 	// encode Data
-	if err := rlp.EncodeString(tx.Data, w, b); err != nil {
+	if err := rlp.EncodeBytes(tx.Data, w, b); err != nil {
```

Comment:
Why this rename? Isn't it understood that "string" here means a flat string of bytes.
Additionally it creates naming discrepancy with `EncodeStringSizePrefix`, `ParseString` and so on.

Reply:
Agree. It was a hard decision. We have two functions named `EncodeString` one of which had to be changed. I couldn't come up with anything better. `EncodeString2` was another candidate, but I didn't like it. Anyways, please drop any ideas.

Reply:
Yeah, maybe we can just get rid of the other one. Till then, `EncodeStringBytes` is a suitable name. I am hoping we can soon truly merge the two sets of implementations for all other methods too.

Reply:
I think it's better to use names `EncodeString` and `EncodeString2` to highlight that it's two flavours of the same thing. And eventually we should merge them.

Reply:
I wish go had function overloading features. I was trying merging them into one function by adding additional parameter and by defining `Writer` interface for a `type SomeType []byte`, but things wasn't looking good